packages/pages/src/vite-plugin/build/closeBundle/functionMetadata.ts

```diff
@@ -61,21 +60,39 @@ const getFunctionMetadataMap = async (
 async function generateFunctionMetadata(
   filepath: string
 ): Promise<[string, FunctionMetadata]> {
-  const buildResult = await esbuild.build({
-    entryPoints: [filepath],
-    outdir: TEMP_DIR,
-    write: false,
-    format: "esm",
-    bundle: true,
-  });
-  const importedFile = await importFromString(buildResult.outputFiles[0].text);
+  project.addSourceFileAtPath(filepath);
+  const defaultExportDeclaration = project
+    .getSourceFile(filepath)
+    ?.getDefaultExportSymbol()
+    ?.getDeclarations()[0];
   const relativePath = path.relative(process.cwd(), filepath);
-
-  if (!importedFile.default) {
-    throw new Error(`${relativePath} does not contain a default export.`);
+  if (Node.isExportAssignment(defaultExportDeclaration)) {
+    const exportIdenfitier = defaultExportDeclaration.getChildrenOfKind(
+      SyntaxKind.Identifier
+    )[0];
+    if (!exportIdenfitier) {
+      throw new Error(
+        `${relativePath} contains a default export assignment that is ` +
+          "improperly formatted. It may have an anonymous function as the default export " +
+          "which is not supported."
+      );
+    }
+    return [relativePath, { entrypoint: exportIdenfitier.getText() }];
+  } else if (Node.isFunctionDeclaration(defaultExportDeclaration)) {
+    const entrypoint = defaultExportDeclaration.getName();
+    if (!entrypoint) {
+      throw new Error(
```

Comment:
Do we know in what case this would happen?

Reply:
updated the docs to more accurately describe this!